### PR TITLE
bitwig-studio5: 5.18 -> 5.19

### PR DESCRIPTION
**Pull request by camelpunch**
The prior URL scheme appears to stop working in this version. Use curl's --location to follow the official link from the website, which redirects.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/audio/bitwig-studio/bitwig-studio5.nix [hunk 1]
```diff
@@ -27,11 +27,12 @@
...
   pname = "bitwig-studio";
-  version = "5.1.8";
+  version = "5.1.9";
 
   src = fetchurl {
-    url = "https://downloads.bitwig.com/stable/${version}/${pname}-${version}.deb";
-    sha256 = "sha256-KxNLae/uTYL1m/X+/7wr7hhKfw31NpB9Mw9RzfrTuus=";
+    url = "https://www.bitwig.com/dl/Bitwig%20Studio/${version}/installer_linux/";
+    curlOpts = "--location";
+    sha256 = "sha256-J5kLqXCMnGb0ZMhES6PQIPjN51ptlBGj4Fy8qSzJ6Qg=";
```
Comment (drupol) [inline pkgs/applications/audio/bitwig-studio/bitwig-studio5.nix:35]:
```suggestion
    hash = "sha256-J5kLqXCMnGb0ZMhES6PQIPjN51ptlBGj4Fy8qSzJ6Qg=";
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/bitwig-studio/bitwig-studio5.nix [hunk 2]
```diff
@@ -27,11 +27,12 @@
 
 stdenv.mkDerivation rec {
   pname = "bitwig-studio";
-  version = "5.1.8";
+  version = "5.1.9";
 
   src = fetchurl {
-    url = "https://downloads.bitwig.com/stable/${version}/${pname}-${version}.deb";
-    sha256 = "sha256-KxNLae/uTYL1m/X+/7wr7hhKfw31NpB9Mw9RzfrTuus=";
+    url = "https://www.bitwig.com/dl/Bitwig%20Studio/${version}/installer_linux/";
```
Comment (drupol) [inline pkgs/applications/audio/bitwig-studio/bitwig-studio5.nix:33]: How about using the canonical `https://downloads-secure.bitwig.com/5.1.9/bitwig-studio-5.1.9.deb` ?

**Comment by camelpunch** [comment]
Can you get that to work without the redirect? It looks like they’ve implemented a form of CSRF protection to ensure you don’t link directly, but I could be wrong.

**Comment by drupol** [comment]
> Can you get that to work without the redirect? It looks like they’ve implemented a form of CSRF protection to ensure you don’t link directly, but I could be wrong.

You're right, I just tested again and the going on the download URL redirected me to the Homepage. Please ignore my feedback on that then.

**Review by drupol — APPROVED** [review]

File: pkgs/applications/audio/bitwig-studio/bitwig-studio5.nix [hunk 3]
```diff
@@ -27,11 +27,12 @@
...
+    curlOpts = "--location";
```
Comment (kkoniuszy) [inline pkgs/applications/audio/bitwig-studio/bitwig-studio5.nix:34]: You don't need to add `--location` to `curlOpts`. `fetchurl` uses it by default anyway: https://github.com/NixOS/nixpkgs/blob/e0ba5d6aaf27ed0deb1ad168bd151a6fd23634f1/pkgs/build-support/fetchurl/builder.sh#L12-L14

Comment (camelpunch) [reply pkgs/applications/audio/bitwig-studio/bitwig-studio5.nix]: Removed - thanks!

**Review by kkoniuszy — APPROVED** [review]

**Comment by drupol** [comment]
Awaiting the feedback from one of the maintainers.

**Review by bfortz — APPROVED** [review]

**Pull request merged**